### PR TITLE
feature(nemesis): add OperatorNodeTerminateAndReplace

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/00UvGVne/2691-k8s-operatornodeterminateandreplace

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Comment by dkropachev** [comment]
Currently blocked by https://github.com/scylladb/scylla-operator/issues/291

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -651,6 +672,49 @@ def _init_new_node_with_latency_calculation(self, old_node_ip, rack=0):
     def run_nodetool_repair_on_new_node(self, node):
         self.repair_nodetool_repair(node)
 
+    def disrupt_terminate_and_recover_node_kubernetes(self):  # pylint: disable=invalid-name
```
Comment (bentsi) [inline sdcm/nemesis.py:675]: Let's printout to the log (info)
Affected:
```
GCE instance -
K8s node        - 
Scylla Pod       X  <-
Scylla node     X
```

Comment (bentsi) [reply sdcm/nemesis.py:675]: go over all possible "terminate" methods in for loop and printout

Comment (dkropachev) [reply sdcm/nemesis.py:650]: done

**Comment by dkropachev** [comment]
Tested - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/dmitry/job/dmitry-operator-byo-4/3

File: sdcm/nemesis.py [hunk 2]
```diff
@@ -650,6 +647,85 @@ def _init_new_node_with_latency_calculation(self, old_node_ip, rack=0):
...
+        self.target_node.mark_to_be_replaced()
+        self._kubernetes_wait_till_node_up_after_been_recreated(self.target_node, old_uid=old_uid)
+
+    def _disrupt_terminate_and_recover_node_kubernetes(self, node, node_terminate_method):  # pylint: disable=invalid-name
+        old_uid = node.k8s_pod_uid
+        node_terminate_method()
+        self._kubernetes_wait_till_node_up_after_been_recreated(node, old_uid=old_uid)
+
+    def _disrupt_terminate_decommission_add_node_kubernetes(self, node, node_terminate_method):  # pylint: disable=invalid-name
+        self._set_current_disruption('TerminateDe')
```
Comment (bentsi) [inline sdcm/nemesis.py:706]: what is De?

**Review by bentsi — APPROVED** [review]

**Pull request merged**